test/testlib/nuodb_database_utilities.go

```diff
@@ -636,6 +642,22 @@ func RunOnNuoDBVersion(t *testing.T, versionCheckFunc func(*semver.Version) bool
 	}
 }
 
+func RunOnNuoDBVersionFromOptions(t *testing.T, options *helm.Options, versionCheckFunc func(*semver.Version) bool, actionFunc func(*semver.Version)) {
```

Comment:
Why not reuse the `RunOnNuoDBVersionFromOptions` in `RunOnNuoDBVersion`?

Reply:
To be more precise you should have something like:

```go
func RunOnNuoDBVersion(t *testing.T, versionCheckFunc func(*semver.Version) bool, actionFunc func(*semver.Version)) {
	RunOnNuoDBVersionFromOptions(t, &helm.Options{}, versionCheckFunc, actionFunc)
}

func RunOnNuoDBVersionFromOptions(t *testing.T, options *helm.Options, versionCheckFunc func(*semver.Version) bool, actionFunc func(*semver.Version)) {
...
}
```

Reply:
If we are actually just calling RunOnNuoDBVersionFromOptions from RunOnNuoDBVersion, should I just not deprecate the RunOnNuoDBVersion ?

Reply:
I have done something like this and extracted the common in both functions.
```
func RunOnNuoDBVersion(t *testing.T, versionCheckFunc func(*semver.Version) bool, actionFunc func(*semver.Version)) {
	versionString := GetNuoDBVersion(t, "default", &helm.Options{})
	// Select only the main NuoDB version (i.e 4.2.1) from the full version string
	checkVersion(t, versionString, versionCheckFunc, actionFunc)
}

func RunOnNuoDBVersionFromOptions(t *testing.T, options *helm.Options, versionCheckFunc func(*semver.Version) bool, actionFunc func(*semver.Version)) {
	versionString, ok := options.SetValues["nuodb.image.tag"]
	if !ok || !NuoDBVersionRegex.MatchString(versionString) {
		versionString = GetNuoDBVersion(t, "default", options)
	}
	checkVersion(t, versionString, versionCheckFunc, actionFunc)
}

func checkVersion(t *testing.T, versionString string, versionCheckFunc func(*semver.Version) bool, actionFunc func(*semver.Version)) {
	version, err := semver.NewVersion(NuoDBVersionRegex.ReplaceAllString(versionString, "${1}"))
	require.NoError(t, err, "Unable to parse NuoDB version", versionString)

	if versionCheckFunc(version) {
		actionFunc(version)
	} else {
		t.Logf("Conditional logic not executed for NuoDB version %s", versionString)
	}
}
```

Reply:
> If we are actually just calling RunOnNuoDBVersionFromOptions from RunOnNuoDBVersion, should I just not deprecate the RunOnNuoDBVersion ?

There is no reason to deprecate `RunOnNuoDBVersion` function as it is just a wrapper of `RunOnNuoDBVersionFromOptions`. 

> I have done something like this and extracted the common in both functions.

You can replace the entire body of the existing `RunOnNuoDBVersion` by calling `RunOnNuoDBVersionFromOptions` with empty options as suggested above.